docs/coding.md

```diff
@@ -0,0 +1,75 @@
+# Coding Conventions
+
+Please follow coding conventions and guidelines described in the following documents:
+
+* [Go proverbs](https://go-proverbs.github.io/) - highly recommended read
+* [CodeReviewComments](https://github.com/golang/go/wiki/CodeReviewComments)
+* [Effective Go](https://golang.org/doc/effective_go.html)
+* [How to Write a Git Commit Message](https://chris.beams.io/posts/git-commit/)
+
+Here's a list of some more specific conventions that are often followed in
+the code and will be pointed out in the review process:
+
+## General
+
+* Keep variable names short for variables that are local to the function.
+* Do not export a function or variable name outside the package until you
+  have an external consumer for it.
+* Do not use named return values in function definitions. Use only the type.
```

Comment:
@Madhu-1 and myself had a disagreement over this, I am fine with the rule being called out, but I would like to know why we should not use named returns and only use naked returns?
The answer of common style across code does not fit IMO as named returns are for short functions and not for all functions as per the language definition, hence all code cannot accommodate a single style.